### PR TITLE
Upgrade Peft version to 0.10.0 for LLM finetune

### PR DESCRIPTION
**Pull request by qiyuangong**
## Description

* Upgrade Peft version to 0.10.0 in HF-Peft example

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [x] Document test
- [ ] ...

### CODE REVIEW

**Comment by qiyuangong** [comment]
Will upgrade peft version in other examples in other pull request.

**Comment by qiyuangong** [comment]
Revert changes in HF-PEFT finetune.py .

**Comment by qiyuangong** [comment]
https://github.com/intel-analytics/ipex-llm/pull/10930

**Comment by qiyuangong** [comment]
Note that this PR will remove support for Peft 0.5.0.

File: python/llm/src/ipex_llm/transformers/qlora.py [hunk 1]
```diff
@@ -62,38 +63,41 @@
...
-            out_features,
-            qtype=kwargs.get("qtype"),
-            bias=kwargs.get("bias", True),
-            conver_to_half=False,
+        super().__init__()
+        LoraLayer.__init__(self, base_layer, **kwargs)
+
+        self.fan_in_fan_out = fan_in_fan_out
+        self._active_adapter = adapter_name
+        self.update_layer(
```
Comment (plusbang) [inline python/llm/src/ipex_llm/transformers/qlora.py:89]: It seems that we don't need to address case of `qa_lora=True` additionally?

Comment (qiyuangong) [reply python/llm/src/ipex_llm/transformers/qlora.py:89]: Peft doesn't add qalora support yet. https://github.com/huggingface/peft/issues/986 We still need to handle qa_lora with our code.

**Review by plusbang — APPROVED** [review]
LGTM

**Pull request merged**